### ISSUE TITLE
Initialize user secrets does not work with older `dotnet user-secrets`

### ISSUE THREAD

**Issue by Witik** [Witik/RiderUserSecrets#21]
I tested the new functionality on my Ubuntu machine, and the initialize action failed without error.

After debugging I discovered the following problems:
1. `dotnet user-secrets --version` (used to check if the tool is installed) returns the following error:
```
Permission denied to modify the '/usr/share/dotnet/sdk/NuGetFallbackFolder' folder.

Here are some options to fix this error:
----------------------------------------
1. Re-run this command with elevated access.
2. Disabled the first run experience by setting the environment variable DOTNET_SKIP_FIRST_TIME_EXPERIENCE to true.
3. Copy the .NET Core SDK to a non-protected location and use it from there.
    
Specify --help for a list of available options and commands.
```
  - This is mitigated by adding `DOTNET_SKIP_FIRST_EXPERIENCE=true` as environment variable to the runtime.
2. The init command returns `Unrecognized command or argument 'init'`
This is due to an older version of the user-secrets tool being installed.
  - This could be solved by checking the version returned by `dotnet user-secrets --version` and adding a "not supported" or "please update" message, when it's older than 3.0.0 (see https://github.com/aspnet/DotNetTools/pull/500)

**Comment by seclerp** [comment]
@Witik what about to manually create `<UserSecretsId>` node in project's XML if not presented for <3.0.0 dotnet CLI tools? It will be a naive implementation but will work in such a case.

Logic could be the same as it is introduced in [real init command](https://github.com/dotnet/aspnetcore/blob/main/src/Tools/dotnet-user-secrets/src/Internal/InitCommand.cs#L106) source code.

**Comment by Witik** [comment]
Yes, that would also work. But then, what's the benefit of using the >=3.0.0 CLI tool? If we have a working alternative for the init command, I'd rather use it in all situations.

**Comment by seclerp** [comment]
It was one of the first questions that I faced when started working on this feature - use of dotnet CLI or own implementation. I prefer to rely on existing tools because in the future they could change their logic and how they apply new secret id, in our example. But you are also right that we could simplify this stuff and go with a single approach. The only drawback here - we need to be in sync with what is going on with `dotnet user-secrets` tool over time.

**Comment by seclerp** [comment]
@Witik I have implemented manual user secrets id generation on my fork, but it needs to be a bit tweaked in terms of formatting. dotnet CLI doesn't respect formatting in `.csproj` files (indentation, primarily) and because of that insertion of new secrets looks ugly in the resulting file using a global tool. On the other hand, in my implementation, I made some work on keeping formatting and using Rider settings for additional indentation. The problem is now in some Java XML-related transformations implementation. For example, it removes spaces between self-closing tags:
`<... />` -> `<.../>` and that adds some redundant changes in Git if the project is in some git repository. If it is okay, I will create PR immediately.

**Comment by seclerp** [comment]
I had a discussion about the ability to use the MSBuild process to change properties from Rider backend, but it leaks some public API calls to get what we want. JetBrains devs told me that it will be available in the 2021.2 release and it will greatly simplify the way how project properties could be managed from the plugin side.
